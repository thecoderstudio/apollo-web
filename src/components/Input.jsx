import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  error: PropTypes.string,
  inverted: PropTypes.bool
};

const defaultProps = {
  error: '',
  inverted: false
};

const Container = styled.div`
  height: 80px;
`;

const StyledInput = styled.input`
  min-height: 50px;
  width: 100%;
  background: ${props => props.inverted ? props.theme.lightBlack : props.theme.white};
  color: ${props => props.inverted ? props.theme.white : "" };
  border: none;
  border-radius: 5px;
  font-family: 'Libre Franklin', sans-serif;
  font-size: 1rem;
  padding-left: 10px;
  opacity: 0.9;
  border: 1px solid ${props => props.hasError ? props.theme.error : 'transparent'};

  &:focus{
    outline: none;
    border: 1px solid ${props => props.theme.accent};
  }
`;

const Error = styled.p`
  color: ${props => props.theme.error};
  margin-top: 5px;
  font-size: 0.9rem;
`;

export default function Input(props) {
  const { className, error, ...rest } = props;
  const hasError = error.length > 0;
  return (
    <Container className={className}>
      <StyledInput hasError={hasError} {...rest} />
      { hasError && <Error>{error}</Error>}
    </Container>
  );
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;
