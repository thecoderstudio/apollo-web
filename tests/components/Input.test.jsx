import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../src/components/Input';

it('renders correctly', () => {
  const tree = renderer
    .create(<Input type="username" placeholder="username" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
