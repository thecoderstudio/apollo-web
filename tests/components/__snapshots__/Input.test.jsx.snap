// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="sc-AxjAm gWnvPB"
  placeholder="username"
  type="username"
/>
`;
